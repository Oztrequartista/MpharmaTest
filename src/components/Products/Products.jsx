import React from "react";
import "./products.css";
import { formattedDate } from "../../UtilityFunctions/myUtils";
import { BsFillTrashFill, BsEyeFill } from "react-icons/bs";
import { BiEdit } from "react-icons/bi";

const Products = ({
  state,
  handleEditAndAdd = () => {},
  handleProductPriceHistory = () => {},
  setIsModalOpen = () => {},
  setDeleteId=()=>{},
  setShowDeleteProducts=()=>{}

}) => {
  return (
    <div className="product-container">
      {state.products.length > 0 &&
        state.products.map((singleProduct, index) => {
          const { itemPrice, name, date, priceId, productId } = singleProduct;
          const dateAsString = formattedDate(date);
          return (
            <div key={index} className="product">
              <h2 className="name">{name.toLowerCase()}</h2>
              <button className="price">GHS {itemPrice}</button>
              <h4 className="date">{dateAsString}</h4>
              <div className="btn-container">
                <div
                  onClick={(event) => {
                    event.preventDefault();
                    handleEditAndAdd(priceId);
                  }}
                  className="btn tooltip"
                  data-testid ="edit"
                >
                  <BiEdit color="#0a0ac0" size={22} />
                  <span className="tooltiptext">Edit Product</span>
                </div>
                <div
                  onClick={(event) => {
                    event.preventDefault();
                    setDeleteId(priceId);
                    setShowDeleteProducts(true);
                  }}
                  className="btn tooltip"
                  data-testid ="filter"
                >
                  <BsFillTrashFill color="#c02626" size={22} />
                  <span className="tooltiptext">Delete Product</span>
                </div>
                <div
                  onClick={(event) => {
                    event.preventDefault();
                    handleProductPriceHistory(productId);
                    setIsModalOpen(true);
                  }}
                  className="btn tooltip"
                  data-testid ="view"
                >
                  <BsEyeFill color="#584c4c" size={22} />
                  <span className="tooltiptext">Previous Prices</span>
                </div>
              </div>
            </div>
          );
        })}
    </div>
  );
};

export default Products;
