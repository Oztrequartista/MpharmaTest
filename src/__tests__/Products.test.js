import React from "react";
import Products from "../components/Products/Products";
import {screen, render, fireEvent} from "@testing-library/react"

describe("Products.js", ()=>{
    it("renders a product card list", ()=>{
         
       const  handleEditAndAdd = jest.fn(); 
       const  handleProductDelete= jest.fn(); 
       const  handleProductPriceHistory= jest.fn(); 
       const setIsModalOpen= jest.fn(); 
       const state = { products : [
        {
            "name": "Exforge 10mg",
            "productId": 5,
            "itemPrice": 10.99,
            "priceId": 8,
            "date": "Sun Apr 03 2022",
            "priceIdArray": [
                8
            ]
        },
    ]};

       const { container } = render(<Products handleEditAndAdd={handleEditAndAdd} handleProductDelete={handleProductDelete} handleProductPriceHistory={handleProductPriceHistory} setIsModalOpen={setIsModalOpen} state={state}/>)

       const edit = screen.getByTestId("edit");
       fireEvent.click(edit);
       const filter = screen.getByTestId("filter");
       fireEvent.click(filter);
       const view = screen.getByTestId("view");
       fireEvent.click(view);

       expect(handleEditAndAdd).toBeCalledTimes(1);

       expect(container.parentElement).toMatchSnapshot();

    })
})