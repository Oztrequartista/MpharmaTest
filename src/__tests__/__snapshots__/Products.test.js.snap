// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Products.js renders a product card list 1`] = `
<body>
  <div>
    <div
      class="product-container"
    >
      <div
        class="product"
      >
        <h2
          class="name"
        >
          exforge 10mg
        </h2>
        <button
          class="price"
        >
          GHS 
          10.99
        </button>
        <h4
          class="date"
        >
          Sun Apr 03 2022
        </h4>
        <div
          class="btn-container"
        >
          <div
            class="btn tooltip"
            data-testid="edit"
          >
            <svg
              color="#0a0ac0"
              fill="currentColor"
              height="22"
              stroke="currentColor"
              stroke-width="0"
              style="color: rgb(10, 10, 192);"
              viewBox="0 0 24 24"
              width="22"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="m7 17.013 4.413-.015 9.632-9.54c.378-.378.586-.88.586-1.414s-.208-1.036-.586-1.414l-1.586-1.586c-.756-.756-2.075-.752-2.825-.003L7 12.583v4.43zM18.045 4.458l1.589 1.583-1.597 1.582-1.586-1.585 1.594-1.58zM9 13.417l6.03-5.973 1.586 1.586-6.029 5.971L9 15.006v-1.589z"
              />
              <path
                d="M5 21h14c1.103 0 2-.897 2-2v-8.668l-2 2V19H8.158c-.026 0-.053.01-.079.01-.033 0-.066-.009-.1-.01H5V5h6.847l2-2H5c-1.103 0-2 .897-2 2v14c0 1.103.897 2 2 2z"
              />
            </svg>
            <span
              class="tooltiptext"
            >
              Edit Product
            </span>
          </div>
          <div
            class="btn tooltip"
            data-testid="filter"
          >
            <svg
              color="#c02626"
              fill="currentColor"
              height="22"
              stroke="currentColor"
              stroke-width="0"
              style="color: rgb(192, 38, 38);"
              viewBox="0 0 16 16"
              width="22"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M2.5 1a1 1 0 0 0-1 1v1a1 1 0 0 0 1 1H3v9a2 2 0 0 0 2 2h6a2 2 0 0 0 2-2V4h.5a1 1 0 0 0 1-1V2a1 1 0 0 0-1-1H10a1 1 0 0 0-1-1H7a1 1 0 0 0-1 1H2.5zm3 4a.5.5 0 0 1 .5.5v7a.5.5 0 0 1-1 0v-7a.5.5 0 0 1 .5-.5zM8 5a.5.5 0 0 1 .5.5v7a.5.5 0 0 1-1 0v-7A.5.5 0 0 1 8 5zm3 .5v7a.5.5 0 0 1-1 0v-7a.5.5 0 0 1 1 0z"
              />
            </svg>
            <span
              class="tooltiptext"
            >
              Delete Product
            </span>
          </div>
          <div
            class="btn tooltip"
            data-testid="view"
          >
            <svg
              color="#584c4c"
              fill="currentColor"
              height="22"
              stroke="currentColor"
              stroke-width="0"
              style="color: rgb(88, 76, 76);"
              viewBox="0 0 16 16"
              width="22"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M10.5 8a2.5 2.5 0 1 1-5 0 2.5 2.5 0 0 1 5 0z"
              />
              <path
                d="M0 8s3-5.5 8-5.5S16 8 16 8s-3 5.5-8 5.5S0 8 0 8zm8 3.5a3.5 3.5 0 1 0 0-7 3.5 3.5 0 0 0 0 7z"
              />
            </svg>
            <span
              class="tooltiptext"
            >
              Previous Prices
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
